File: 3300-minimum-element-after-replacement-with-digit-sum/3300-minimum-element-after-replacement-with-digit-sum.go
func minElement(nums []int) int {
    ans := math.MaxInt
    for _, num := range nums {
        curr := 0;
        for num > 0 {
            curr += num % 10
            num /= 10
        }
        ans = min(ans, curr)
    }     
    return ans
}

File: 3300-minimum-element-after-replacement-with-digit-sum/3300-minimum-element-after-replacement-with-digit-sum.rs
impl Solution {
    pub fn min_element(nums: Vec<i32>) -> i32 {
        let mut ans = i32::MAX;
        for num in nums {
            let mut num = num;
            let mut curr = 0;
            while num > 0 {
                curr += num % 10;
                num /= 10;
            }
            ans = ans.min(curr);
        } 
        ans
    }
}